src/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Windows.cs

```diff
@@ -27,6 +27,18 @@ public static bool IsProcessRunning(int processId)
         /// <returns>true if the process is running; otherwise, false.</returns>
         public static bool IsProcessRunning(int processId, string machineName)
         {
+            //Performance optimization for the local machine: 
+            //First try to OpenProcess by id, if valid handle is returned, the process is definitely running
+            //Otherwise enumerate all processes and compare ids
+            if (!IsRemoteMachine(machineName))
+            {
+                SafeProcessHandle processHandle = Interop.Kernel32.OpenProcess(Interop.Advapi32.ProcessOptions.PROCESS_QUERY_INFORMATION, false, processId);
+                if (!processHandle.IsInvalid)
+                {
+                    return true;
+                }
```

Comment:
The processHandle should be in a using block so it's disposed of.  Otherwise this ends up leaving the SafeHandle for finalization.